src/lib.rs

```diff
@@ -275,6 +276,36 @@ impl Pixels {
         self.context.scaling_renderer.clear_color = color;
     }
 
+    /// Returns a reference of the `wgpu` adapter used by the crate.
+    /// 
+    /// The adapter can be used to retrieve runtime information about the host system
+    /// or the WGPU backend.
+    /// ```no_run
+    /// use pixels::Pixels;
+    /// 
+    /// let mut pixels = Pixels::new(320, 240, surface_texture)?;
+    /// let adapter = pixels.get_wgpu_adapter();
+    /// // Do something with the adapter.
+    /// ```
+    pub fn get_wgpu_adapter(&self) -> &wgpu::Adapter {
+        &self.adapter
+    }
+
+    /// Returns a mutable reference to the `wgpu` adapter used by the crate.
+    /// 
+    /// The adapter can be used to retrieve runtime information about the host system
+    /// or the WGPU backend.
+    /// ```no_run
+    /// use pixels::Pixels;
+    /// 
+    /// let mut pixels = Pixels::new(320, 240, surface_texture)?;
+    /// let adapter = pixels.get_wgpu_adapter();
+    /// // Do something with the adapter.
+    /// ```
+    pub fn get_wgpu_adapter_mut(&mut self) -> &mut wgpu::Adapter {
+        &mut self.adapter
```

Comment:
I don't think this is needed, since all `wgpu::Adapter` methods only take `&self`. And I suspect we don't want to allow things like `std::mem::swap()` to be used here.